docs/docs/developer/frontend/style-guide.mdx

```diff
@@ -153,6 +151,25 @@ let color = Color.Red;
 let color: "red" | "green" | "blue" = "red";
 ```
 
+#### GraphQL and internal libs
+
+We recommend using enums that are generated by GraphQL codegen.
+
+We also recommend using an enum when using an internal lib, so the internal lib doesn't have to expose a string literal type that is not related to the internal API.
+
+Example : 
+
+```TSX
+const {
+  setHotkeyScopeAndMemorizePreviousScope,
+  goBackToPreviousHotkeyScope,
+} = usePreviousHotkeyScope();
+
+setHotkeyScopeAndMemorizePreviousScope(
```

Comment:
mmmh ok, good one! I don't think it's clear enough:
I would put create two differents paragraphs, 1 for Graphql and 1 for internal libraries use case.

In the internal lib use case, I would take more time to expose the issue: 
- in some cases, we want to be able to pass a string amongst a list of string to a library (ex. setHotkeyScopeAndMemorizePreviousScope that can take "relation-picker" | "companies-table" as a valid scope)
- however, the library code should not be aware of this specific list which is only a concern for the library api user. We don't want developers to add a new possibility to the library each time they need it. This means the library will accept any string as a valid input. setHotkeyScopeAndMemorizePreviousScope(scope: string)
- in this case, we use enums to make sure that we are passing an authorized scope to the function (RelationPickerHotkeyScope.RelationPicker | AppHotkeyScope.CompaniesTable)